packages/api/src/graphql/query.private.ts

```diff
@@ -290,6 +293,57 @@ export const GraphQLQuery = new GraphQLObjectType<undefined, Context>({
       }
     },
 
+    userAndSubscriptionBulkData: {
+      type: GraphQLString,
+      args: {},
+      async resolve(root, args, {dbAdapter, authenticate}) {
+        const {roles} = authenticate()
+
+        authorise(CanExportUserAndSubscription, roles)
+
+        const users = await dbAdapter.user.getUsers({
+          cursor: InputCursor(),
+          filter: {subscription: {}},
+          limit: Limit(1000),
+          sort: UserSort.ModifiedAt,
+          order: SortOrder.Descending
+        })
+
+        if (users?.nodes.length) {
+          const list: UserWithSubscription[] = users.nodes.map(
```

Comment:
Here it looks like your doing the work twice. First you map through the users and then pass the array to the `parse` function which i assume will do another loop through all the objects. 

I suggest you get rid of the `parse` function and remove the dependency from the package.json. Instead of `map` i suggest you use `reduce` and for every users you add a new CSV line. Something like:
```
users.reduce((previousValue, currentValue) => {
  previousValue = `${previousValue}\n${currentValue.id},${currentValue.createdAt},....`
  return previousValue
}, ""}
```
Also you don't need the new type `UserWithSubscription`

Reply:
can you check again?